rest/client.go

```diff
@@ -156,6 +167,26 @@ func (c Client) Do(req *http.Request, v interface{}) (*http.Response, error) {
 	return resp, err
 }
 
+// nextFunc knows how to get and parse additional info from uri into v.
+type nextFunc func(v *interface{}, uri string) (*http.Response, error)
+
+// DoWithPagination Does, and follows Link headers for pagination.
+func (c Client) DoWithPagination(req *http.Request, v interface{}, f nextFunc) (*http.Response, error) {
```

Comment:
You have mixed exposed and unexposed types in this signature. I'd pick one and stick with it. However, is there any strong reason why this function should be exposed at all?

Reply:
good point. `Do` could probably also be unexposed. But it could be nice - from an SDK as toolkit perspective - to let users get at some of the guts if they want do do something fancy.

For consistency and least amount of change, inclined to keep `DoWithPagination` exposed and uppercase `NextFunc`